keps/sig-storage/177-volume-snapshot/README.md

```diff
@@ -231,24 +251,288 @@ Here are the changes since the original design proposal:
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+  checking if there are objects with field X set) may be a last resort. Avoid
+  logs or events for this purpose.
+  Metrics support is already added to CSI external-snapshotter sidecar. Metrics support will be added to the snapshot-controller in 1.20.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [x] Metrics
+    - Metric name: snapshot_operation_total_seconds, snapshot_operation_count, snapshot_csi_total_seconds
+    - [Optional] Aggregation method:
+    - Components exposing the metric: snapshot-controller and CSI external-snapshotter sidecar
+  - [ ] Other (treat as last resort)
+    - Details:
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  At a high level, this usually will be in the form of "high percentile of SLI
+  per day <= X". It's impossible to provide comprehensive guidance, but at the very
+  high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99,9% of /health requests per day finish with 200 code
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+  Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+  implementation difficulties, etc.).
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+  Think about both cluster-level services (e.g. metrics-server) as well
+  as node-level agents (e.g. specific version of CRI). Focus on external or
+  optional services that are needed. For example, if this feature depends on
+  a cloud provider API, or upon an external software-defined storage or network
+  control plane.
+
+  For each of these, fill in the following—thinking about running existing user workloads
+  and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]: installation of snapshot CRDs, snapshot-controller, snapshot validation webhook, CSI external-snapshotter sidecar
+    - Usage description:
+      - Impact of its outage on the feature: installation of snapshot CRDs, snapshot-controller, and CSI external-snapshotter sidecar are required for the volume snapshot feature to work. Snapshot validation webhook is optional. If the validation webhook is not running, API validation will not happen.
```

Comment:
What happens if we have invalid API objects?  Also, if the resources become invalid, is it really optional?

Reply:
The invalid API objects are not usable. We have logic in the controller to check that.  That logic was added before the validation web hook was implemented.  Since the validation web hook, the snapshot controller, and the snapshot CRDs are all out-of-tree, I'm not sure that we have a way to make the web hook required.  We recommend K8S distro to install them.